### ISSUE TITLE
Change isItemSelecetd to a value rather than a function

### ISSUE THREAD

**Issue by jakec-dev** [jakec-dev/travel-planner#32]
@HcroakerDev Hey, I've got a question for you.

What are the implications of using the current implementation of `isItemSelected`:

https://github.com/jakec-dev/travel-planner/blob/20b2e4c826a5952405de7b1b113ffbb198384a66/src/components/GearTable/Item/Item.jsx#L22-L27

https://github.com/jakec-dev/travel-planner/blob/20b2e4c826a5952405de7b1b113ffbb198384a66/src/components/GearTable/Item/Item.jsx#L32-L36

compared to changing it to a value like so:

```js
const isItemSelected = useMemo(
  () =>
    selectedItems.findIndex((selectedItem) => selectedItem === item.id) >= 0,
  [selectedItems]
);

// ....
<input
  type="checkbox"
  checked={isItemSelected}
  onChange={handleSelect}
/>
```

My thoughts are that they would both effectively work in the same way and either method would be acceptable, however the second method is more readable so that would be preferred.

A third method to handle this would be:

```js
const checkItemSelected = (itemId) => {
  const itemIndex = selectedItems.findIndex(
    (selectedItem) => selectedItem === itemId
  );
  return itemIndex >= 0;
};

const isItemSelected = useMemo(
  () => checkItemSelected(item.id),
  [selectedItems]
);

// ....
<input
  type="checkbox"
  checked={isItemSelected}
  onChange={handleSelect}
/>
```
However this just seems needlessly complex in my opinion - correct me if I'm wrong.

And a fourth method would be:

```js
const isItemSelected = useMemo(
  () =>
    (() => {
      const itemIndex = selectedItems.findIndex(
        (selectedItem) => selectedItem === item.id
      );
      return itemIndex >= 0;
    })(),
  [selectedItems]
);

// ....
<input
  type="checkbox"
  checked={isItemSelected}
  onChange={handleSelect}
/>
```
Which is even more confusing than the third method and pretty obviously the worst option of the lot.

So basically, would I be correct in saying that the second method would be the preferred option here? The only real reason to use the first (or third) method instead would be if other actions needed to be run while performing the check (e.g. throwing in a console.log or something before returning the value), but since all this function does is return a value then the second method is a better choice.
